docs/tutorials/rerank/rerank_pipeline_with_CrossEncoder_model_deployed_on_Sagemaker.md

```diff
@@ -0,0 +1,379 @@
+# Topic
+
+[Reranking pipeline](https://opensearch.org/docs/latest/search-plugins/search-relevance/reranking-search-results/) is a feature released in OpenSearch 2.12.
+It can rerank search results, providing a relevance score for each document in the search results with respect to the search query.
+The relevance score is calculated by a cross-encoder model. 
+
+This tutorial explains how to use the [Huggingface cross-encoder/ms-marco-MiniLM-L-6-v2](https://huggingface.co/cross-encoder/ms-marco-MiniLM-L-6-v2) model in a reranking pipeline. 
+
+Note: Replace the placeholders that start with `your_` with your own values.
+
+# Steps
+
+## 0. Deploy Model on Sagemaker
+Use this code to deploy model on Sagemaker.
+```python
+import sagemaker
+import boto3
+from sagemaker.huggingface import HuggingFaceModel
+sess = sagemaker.Session()
+role = sagemaker.get_execution_role()
+
+hub = {
+    'HF_MODEL_ID':'cross-encoder/ms-marco-MiniLM-L-6-v2',
+    'HF_TASK':'text-classification'
+}
+huggingface_model = HuggingFaceModel(
+    transformers_version='4.37.0',
+    pytorch_version='2.1.0',
+    py_version='py310',
+    env=hub,
+    role=role, 
+)
+predictor = huggingface_model.deploy(
```

Comment:
Can we also add instruction for GPU usage? Since we have batch ingestion now, ingestion throughput can benefit a lot from using GPU. 
Customer can choose GPU container from https://github.com/aws/deep-learning-containers/blob/master/available_images.md#huggingface-training-containers by using proper transformer + pytorch + py version and set a GPU instance like g4dn/g5.xlarge. Then the endpoint will use GPU for inference automatically.

Reply:
Let's keep this tutorial focusing on current topic. We can create a separate tutorial for GPU usage.

Reply:
Sorry for late reply. I miss this message. We also want to create some docs for tutorial of neural sparse model. Maybe we will raise them together later.